include/cvc5/cvc5_parser.h

```diff
@@ -268,6 +268,76 @@ class CVC5_EXPORT InputParser
   std::shared_ptr<Parser> d_parser;
 };
 
+/**
+ * Base class for all Parser exceptions.
+ * If thrown, API objects can still be used
+ */
+class CVC5_EXPORT ParserException : public CVC5ApiException
+{
+ public:
+  // Constructors
+  ParserException();
+
+  ParserException(const std::string& msg);
+
+  ParserException(const char* msg);
+
+  ParserException(const std::string& msg,
+                  const std::string& filename,
+                  unsigned long line,
+                  unsigned long column);
+
+  // Destructor
+  ~ParserException() override;
+
+  void toStream(std::ostream& os) const override;
+
+  std::string getFilename() const;
+
+  /** get the line number of the parsing error */
+  unsigned long getLine() const;
+  /** get the column number of the parsing error */
+  unsigned long getColumn() const;
+
+ protected:
+  std::string d_filename;
+  unsigned long d_line;
+  unsigned long d_column;
+}; /* class ParserException */
+
+/**
+ * An end of file exception.
+ * If thrown, API objects can still be used
+ */
+class ParserEndOfFileException : public ParserException
```

Comment:
It seems that this is not used, why was this introduced? I suppose, as discussed offline, this can be removed.

Reply:
I think we need to keep this, for https://github.com/cvc5/cvc5/blob/main/src/main/interactive_shell.cpp#L375.

Discussed offline, I think it is fine to introduce this into the API.